### DISCUSSION TITLE
Use opentelemetry-java without ThreadLocalStorage

### DISCUSSION THREAD

**Discussion by johnpoth** [open-telemetry/opentelemetry-java#6683] [Q&A]
Hey guys awesome project!

I was wondering if there was a way to use this library without relying on all the ThreadLocalContextStorage stuff. We have a lot of Threads in our project and it is a nightmare to debug when scopes aren't closed in the same thread.

I feel like https://opentelemetry.io/docs/languages/java/instrumentation/#create-nested-spans is the closest I've found but there is a call to Context.current() so not quite :)

Thanks !

**Comment by trask** [comment]
(linking to https://github.com/open-telemetry/opentelemetry-java/discussions/6584#discussioncomment-10547812)
